.github/workflows/build_docker.yml

```diff
@@ -0,0 +1,100 @@
+name: "Build Docker"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    branches:
+      - "main"
+    paths-ignore:
+      - "**.md"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      CACHE_IMAGE: "ghcr.io/vllm-project/vllm-spyre:build-cache"
+      CACHE_REGISTRY: "ghcr.io"
+      QUAY_REPOSITORY: "quay.io/ibm-aiu/vllm-spyre"
+
+    permissions:
+      packages: write
+      contents: read
+
+    steps:
+    - name: "Checkout"
+      uses: actions/checkout@v4
+
+    - name: "Set up QEMU"
+      uses: docker/setup-qemu-action@v3
+
+    - name: "Set up Docker Buildx"
+      uses: docker/setup-buildx-action@v3
+
+    - name: "Log in to Github Container registry"
+      uses: docker/login-action@v3
+      if: github.event_name != 'pull_request'
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
```

Comment:
I am not 100% sure, but this might work more reliably using `username: ${{ github.repository_owner }}` and a his/her secret (setup in the repo settings) so people without write privileges can make use of the GHCR cache

Reply:
🌶️  Will look into it a bit

Reply:
This would be hype, but atm this image build is _super fast_

I think the PPC and Z builds, if we integrate them here, will be slower and could use shared cache for external users so we should look at it then